src/kak-setsid.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <unistd.h>
+
+void execute_command_in_arguments(int argc, char *argv[]);
+
+int main(int argc, char *argv[])
+{
+    if(argc < 2) {
+        fprintf(stderr, "Give a command to execute as argument to %s.\n",
+                argv[0]);
+        return 1;
+    }
+    
+    switch(fork()) {
+        case -1:
+            fprintf(stderr, "Failed to create child process.\n");
+            return 2;
+        case 0:
+            setsid();
+            execute_command_in_arguments(argc, argv);
+            return 3;  /* execute_command_in_arguments() should not return. */
+        default:
+            break;
+    }
+}
+
+void execute_command_in_arguments(int argc, char *argv[])
+{
+    int i;
+    char *command[argc];
```

Comment:
huh, variable-sized automatic arrays got into C99?  Neat.

Reply:
According to [this](https://retrocomputing.stackexchange.com/questions/5179/why-historically-include-the-number-of-arguments-argc-as-a-parameter-of-main), ANSI C already NULL-terminates the argv array.  This means that, instead of creating `command` and copying to it, you can just call `execvp(argv[1], argv + 1)`.

Reply:
Nice, that let me make the code much more compact.